pkg/operator/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_runtime_reconcile.go

```diff
@@ -74,16 +80,37 @@ func (c *runtimeReconcile) reconcile(ctx context.Context, cm *operatorapiv1.Clus
 		}
 	}
 
+	// Remove grpc server deployment if the grpc auth is disabled
+	if !config.GRPCAuthEnabled {
+		_, _, err := cleanResources(ctx, c.kubeClient, cm, config, grpcServerDeploymentFiles...)
+		if err != nil {
+			return cm, reconcileStop, err
+		}
+	}
+
 	if cm.Spec.RegistrationConfiguration != nil && cm.Spec.RegistrationConfiguration.RegistrationDrivers != nil {
 		var enabledRegistrationDrivers []string
 		for _, registrationDriver := range cm.Spec.RegistrationConfiguration.RegistrationDrivers {
 			enabledRegistrationDrivers = append(enabledRegistrationDrivers, registrationDriver.AuthType)
-			if registrationDriver.AuthType == commonhelpers.AwsIrsaAuthType && registrationDriver.AwsIrsa != nil {
-				config.HubClusterArn = registrationDriver.AwsIrsa.HubClusterArn
-				config.AutoApprovedARNPatterns = strings.Join(registrationDriver.AwsIrsa.AutoApprovedIdentities, ",")
-				config.AwsResourceTags = strings.Join(registrationDriver.AwsIrsa.Tags, ",")
-			} else if registrationDriver.AuthType == commonhelpers.CSRAuthType && registrationDriver.CSR != nil {
-				config.AutoApprovedCSRUsers = strings.Join(registrationDriver.CSR.AutoApprovedIdentities, ",")
+			switch registrationDriver.AuthType {
+			case commonhelpers.AwsIrsaAuthType:
+				if registrationDriver.AwsIrsa != nil {
+					config.HubClusterArn = registrationDriver.AwsIrsa.HubClusterArn
+					config.AutoApprovedARNPatterns = strings.Join(registrationDriver.AwsIrsa.AutoApprovedIdentities, ",")
+					config.AwsResourceTags = strings.Join(registrationDriver.AwsIrsa.Tags, ",")
+				}
+			case commonhelpers.CSRAuthType:
+				if registrationDriver.CSR != nil {
+					config.AutoApprovedCSRUsers = strings.Join(registrationDriver.CSR.AutoApprovedIdentities, ",")
+				}
+			case commonhelpers.GRPCCAuthType:
+				if registrationDriver.GRPC == nil {
+					config.GRPCServerImage = defaultGRPCServerImage
```

Comment:
when it is not set, it should use registration image defined in the clustermanager. So it aligns with the released version.